packages/content/src/types.ts

```diff
@@ -0,0 +1,137 @@
+import { IItem, IGroup } from "@esri/arcgis-rest-types";
+
+export type HubType = 'document' | 'dataset' | 'map' | 'app' | 'site' | 'initiative' | 'event' | 'template';
+
+// Based on the [Hub Content UI Inventory](https://app.lucidchart.com/invitations/accept/dab82a25-2ed9-4f63-8075-3db8873087e3)
+// Attempting a simple _Hungarian Notation_ for attributes suffix: `*Date`, `*Id`, `*Number`. Plural (e.g. `tags`) is an array.
+
+export interface IContent extends IItem {
+  
+  // Inherited metadata (noting relevant subset of all item attributes)
+  // id: string; // item.id
+  // title: string; // item.title
+  // snippet: string; // item.snippet
+  // description: string; // item.description
+  // categories?: Array<string>; // item.categories
+  // tags: Array<string>; // item.tags  
+  // culture: string; // written locale e.g. en, es, cn
+  // url?: string;
+  // size: number;
+
+  // Derived metadata
+  hubType: HubType;
+  maintainer: IHubMember; // item.owner with more user metadata
+  access: { // overrides item.access with more attributes. could flatten.
+    visibility: 'private' | 'org' | 'public'; // item.access
+    permission?: 'view' | 'edit' | 'admin'; // item.itemControl
+    groups?: Array<IGroup>; // item.sharing.groups via content/users/:username/items/:id
+  };
+
+  // Explicit data information since this is a common confusion + bug report
+  createdDate: Date; // formal metadata || new Date(item.created)
+  createdDateSource: string; // description of what was used for this attribute
+  publishedDate: Date; // formal metadata || new Date(item.created)
```

Comment:
what's the difference btw this and `createdDate`?

Reply:
Dataset measured January 15-March 24 (so created March 24)
Dataset processed + finally pushed to web April 10
Dataset data updated to fix bad valued on September 9